lib/pltr/v2/actions/__tests__/ui.test.js

```diff
@@ -0,0 +1,125 @@
+import { removeSystemKeys } from '../../reducers/systemReducers'
+import {
+  cardDialogBeatIdSelector,
+  cardDialogCardIdSelector,
+  cardDialogLineIdSelector,
+  isCardDialogVisibleSelector,
+} from '../../selectors/ui'
+import { emptyFile } from '../../store/newFileState'
+import { loadFile, setCardDialogClose, setCardDialogOpen } from '../ui'
+import { configureStore } from './fixtures/testStore'
+
+const EMPTY_FILE = emptyFile('Test file')
+
+const initialStore = () => {
+  const store = configureStore()
+  store.dispatch(
+    loadFile(
+      'Test file',
+      false,
+      EMPTY_FILE,
+      EMPTY_FILE.file.version,
+      'device://tmp/dummy-url-test-file.pltr'
+    )
+  )
+  return store
+}
+
+const initialCardDialogState = {
+  isOpen: false,
+  cardId: null,
+  lineId: null,
+  beatId: null,
+}
+
+describe('cardDialog', () => {
+  const store = initialStore()
+  const initialState = removeSystemKeys(store.getState().present)
+  describe('given the initial state store', () => {
+    const cardId = cardDialogCardIdSelector(initialState)
+    const beatId = cardDialogBeatIdSelector(initialState)
+    const lineId = cardDialogLineIdSelector(initialState)
+    const isOpen = isCardDialogVisibleSelector(initialState)
+    it('should produce initial state store', () => {
+      expect(initialState.ui.cardDialog).toMatchObject(initialCardDialogState)
+    })
+
+    it('should have `null` as initial values for cardId lineId and beatId', () => {
+      expect(cardId).toBeNull()
+      expect(beatId).toBeNull()
+      expect(lineId).toBeNull()
+    })
+
+    it('should have false as initial valuefor isOpen', () => {
+      expect(isOpen).toBeFalsy()
+    })
+  })
+
+  describe('given cardDialog actions are dispatched', () => {
+    describe('given setCardDialogOpen is dispatched', () => {
+      store.dispatch(setCardDialogOpen(1, 1, 1))
```

Comment:
Does a card with `id` 1 exist?
I think `pltr` should prevent the card from being opened if it doesn't exist.
You can handle that in either the main reducer (`lib/pltr/v2/reducers/main.js`) or by using `redux-thunk` (see `editFileName` in `actions/ui.js`.)
In this case, I think the appropriate choice is probably to change your action because an action should never be dispatched if the card doesn't exist.

Reply:
If you did all that, then you'd need to change the expectations of this test and add another test where we first add a card before manipulating it.